packages/vue/src/components/JsonForms.vue

```diff
@@ -189,29 +189,23 @@ export default defineComponent({
   },
   watch: {
     schema(newSchema) {
-      const generatorData = isObject(this.data) ? this.data : {};
-      this.schemaToUse = newSchema ?? Generate.jsonSchema(generatorData);
+      this.schemaToUse = newSchema ?? Generate.jsonSchema(this.dataToUse);
       if (!this.uischema) {
-        this.uischemaToUse = Generate.uiSchema(
-          this.schemaToUse,
-          undefined,
-          undefined,
-          this.schemaToUse
-        );
+        this.uischemaToUse = generateUISchema(this.schemaToUse);
       }
     },
     uischema(newUischema) {
-      this.uischemaToUse =
-        newUischema ??
-        Generate.uiSchema(
-          this.schemaToUse,
-          undefined,
-          undefined,
-          this.schemaToUse
-        );
+      this.uischemaToUse = newUischema ?? generateUISchema(this.schemaToUse);
     },
     data(newData) {
-      this.dataToUse = newData;
+      this.dataToUse = newData === undefined ? {} : newData;
```

Comment:
Why is this necessary? For instance, we don't do this in React

Reply:
since we do generate the json schema based on the data when the jsonschema is not provided then if the data is undefined then the whole jsonforms will throw an error so this update of the data follows what we have initially on the data property initialization. If jsonforms works works without errors with undefined data when not providing the jsonschema then that code can be removed